src/Piece/Knight.cpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+#include "Piece/Knight.h"
+
+bool Knight::isValidMove(const Position src, const Position des) const {
+    vector<Position> moves = {
+            {src.row + 2, src.col + 1},
+            {src.row + 2, src.col - 1},
+            {src.row - 2, src.col + 1},
+            {src.row - 2, src.col - 1},
+            {src.row + 1, src.col + 2},
+            {src.row + 1, src.col - 2},
+            {src.row - 1, src.col + 2},
+            {src.row - 1, src.col - 2}
+    };
+
+    for (auto move : moves)
```

Comment:
#pragma once
#include "Piece/Pawn.h"

bool Pawn::isValidMove(const Position src, const Position des) const {
    if(isFirstMove)
        if(this->isWhite())
            return (des.row == src.row + 2 || des.row == src.row + 1) && des.col == src.col;
        else
            return (des.row == src.row - 2 || des.row == src.row - 1) && des.col == src.col;
    else
        if(this->isWhite())
            return des.row == src.row + 1 && des.col == src.col;
        else
            return des.row == src.row - 1 && des.col == src.col;

}

ChessPieceType Pawn::getType() const {
    return ChessPieceType::PAWN;
}

void Pawn::setFirstMove() {
    isFirstMove = false;
}